src/matlab2tikz.m

```diff
@@ -998,7 +998,9 @@ function addCustomCode(fid, before, code, after)
     backgroundColor = get(handle, 'Color');
     if ~isNone(backgroundColor)
         [m2t, col] = getColor(m2t, handle, backgroundColor, 'patch');
-        if ~strcmp(col, 'white')
+        % Fill background if not just white or if multiple axes and current
+        % is visible
+        if ~strcmp(col, 'white') || (numel(m2t.relevantAxesHandles) > 1 && isVisible(handle))
```

Comment:
@okomarov I agree with you that this fix is more important than the edge case I saw (because it is only visible when the background is not white, which should only happen on posters or beamer slides, as far as I know and the fix is not possible in a one-liner).